kwarchart/src/main/java/com/kwarchart/android/chart/BarChart.kt

```diff
@@ -1,18 +1,277 @@
+            mMaxLen = it.data.size
+        }
+
+        it.data.forEach { chartData ->
+            if (mMaxVal < chartData.value) {
+                mMaxVal = chartData.value
+            }
+
+            if (!keys.contains(chartData.key)) {
+                keys.add(chartData.key)
+            }
+        }
+    }
+
+    Chart(
+            modifier = modifier,
+            xAxisName = xAxisName,
+            yAxisName = yAxisName,
+            legend = legendPos?.let { _ ->
+                val tmpLegend = Legend(legendPos)
+                data.forEach {
+                    tmpLegend.legends.add(it.legend)
+                    tmpLegend.colors.add(it.color)
+                }
+                tmpLegend
+            }
+    ) {
+        Canvas(
+                modifier = modifier
+                        .weight(5f)
+                        .padding(
+                                start = (mMaxVal.toString().length * 8).dp,
+                                top = 10.dp,
+                                end = 10.dp,
+                                bottom = AXIS_VALUES_FONT_SIZE.dp
+                        )
+        ) {
+            mHGap = size.width / mMaxLen
+            mVGap = size.height / mMaxLen
+
+            if (showGrid) {
+                drawGrids(mMaxLen, gridsColor)
+            }
+            if (showAxes) {
+                drawAxes(axesColor, keys)
+            }
+            data.forEach {
+                drawData(it)
+            }
+        }
+    }
+}
+
+
+/**
+ * Plot line chart data.
+ *
+ * @param barSeries Data to be plotted in this chart.
+ */
+private fun <T> DrawScope.drawData(barSeries: BarSeries<T>) {
+    barSeries.data.forEachIndexed { i, chartData ->
+        val point = getDataPoint(i, chartData, size.height)
+        drawRect(
+                topLeft = Offset(point.x - 24, point.y),
+                size = Size(width = 48f, height = size.height - point.y),
+                color = barSeries.color,
+                style = Stroke(barSeries.width.toFloat()),
+//              brush = Brush.linearGradient(colors = listOf(barSeries.areaColor!!))
+        )
+    }
+
+}
+
+/**
+ * Get position of data point to be plotted in chart.
+ *
+ * @param index Data index.
+ * @param data Chart data.
+ * @param canvasHeight Canvas' height.
+ *
+ * @return Data point's position in canvas.
+ */
+private fun <T> getDataPoint(
+        index: Int,
+        data: ChartData<T>,
+        canvasHeight: Float
+) = Offset(
+        (index + 1) * mHGap,
+        canvasHeight - ((data.value / mMaxVal) * canvasHeight)
+)
+
+/**
+ * Canvas' origin point.
+ *
+ * @return Offset point.
+ */
+private fun DrawScope.origin() = Offset(0f, size.height)
+
+/**
+ * Draw line chart grids.
+ *
+ * @param count Grid count.
+ * @param color Grid color.
+ */
+private fun DrawScope.drawGrids(count: Int, color: Color) {
+    val startPoint = origin()
+//    val pathEffect = PathEffect.dashPathEffect(intervals = floatArrayOf(10f, 10f))
+
```

Comment:
remove if not used

Reply:
remove all commented codes if not used

Reply:
Let me handle this one... Will fix it in HORIZONTAL BarChart PR.